client/src/model/values/instantaneousValues/vectorValues/VectorValue.hx

```diff
@@ -0,0 +1,148 @@
+package model.values.instantaneousValues.vectorValues;
+import model.values.instantaneousValues.vectorValues.VectorValueI;
+import model.values.instantaneousValues.InstantaneousValueI;
+import model.values.instantaneousValues.scalarValues.ScalarValueI;
+
+/**
+ * ...
+ * @author AdvaitTrivedi
+ */
+class VectorValue implements VectorValueI 
+{
+	var vector : Array<InstantaneousValueI> = new Array<InstantaneousValueI>();
+
+	public function new() 
+	{
+		
+	}
+	
+	
+	/* INTERFACE model.values.VectorValueI */
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function fromArray(instanteousValues: Array<InstantaneousValueI>) : VectorValueI {
+		for (value in instanteousValues) {
+			this.push(value);
+		}
+		
+		return this;
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function push(instantaneousValue:InstantaneousValueI, ?index:Int = 0):Void 
```

Comment:
If you keep it, please rename to insert.  (Push on an array inserts at most significant end.)

Reply:
Will need this to create a public mutator, i.e. SignalValue, which will be upcoming in a commit.

Reply:
Renaming it to `insert()`

Reply:
See comments above on mutators.